openssl/src/cms.rs

```diff
@@ -0,0 +1,63 @@
+//! CMS archive
+
+use ffi;
+use foreign_types::{ForeignType, ForeignTypeRef};
+use std::ptr;
+use error::ErrorStack;
+
+use bio::{MemBio, MemBioSlice};
+
+use x509::X509;
+use pkey::PKeyRef;
+
+use cvt;
+use cvt_p;
+
+#[cfg(not(libressl))]
+foreign_type! {
+    type CType = ffi::CMS_ContentInfo;
+    fn drop = ffi::CMS_ContentInfo_free;
+
+    pub struct Cms;
```

Comment:
Do you think these should be called `Cms` or `CmsContentInfo`? I usually like to use the OpenSSL name when possible.